crates/provider-http-server/README.md

```diff
@@ -12,16 +12,87 @@ wash build -p src/bin/http-server-provider
 
-The configuration passed to this provider at startup primarily defines the "mode" the HTTP server should be running in. The `address` mode sets up a listener on a provided address for **each** linked component.
+The configuration passed to this provider at startup primarily defines the "mode" the HTTP server should be running in. The `address` mode sets up a listener on a provided address for **each** linked component, and the `path` mode sets up a listener on a provided address, using the `path` link configuration to route to the linked component.
 
 | Key               | Value                  | Default        | Description                                                                                                           |
 | ----------------- | ---------------------- | -------------- | --------------------------------------------------------------------------------------------------------------------- |
 | `routing_mode`    | "address"              | `address`      | Dictates the routing mode of the capability provider. `address` mode will listen on a new address for each component. |
 | `default_address` | A valid listen address | "0.0.0.0:8000" | The default listen address to listen on and route to components.                                                      |
 
-## Link Configuration
+Configuration differs slightly depending on the `routing_mode` chosen for the HTTP server.
+
+### Address routing mode
+
+In address routing mode, the above configuration of `routing_mode` and `default_address` is supplied as provider configuration. Each [link](https://wasmcloud.com/docs/concepts/linking-components/linking-at-runtime/) from the HTTP server to a component can be configured using the [HTTP address configuration](#http-address-configuration) below. Since the `default_address` can only be used once, the `address` field is a **required** configuration value for any deployments serving more than one component.
+
+### Path routing mode
+
+In path routing mode, the above configuration of `routing_mode` and `default_address` is supplied as provider configuration as well as all values in [HTTP address configuration](#http-address-configuration). The HTTP server, when in path routing mode, sets up a listener at startup to serve **all** components.
+
+All components must be configured with a `path` on the link config for routing in this mode.
+
+| Key    | Default | Description                                                                                                    |
+| ------ | ------- | -------------------------------------------------------------------------------------------------------------- |
+| `path` | `N/A`   | **Required.** The path, e.g. `/api/v1`, to register to send all requests at that path to the linked component. |
```

Comment:
I think it would probably be good to also require a host_name value here. Otherwise a path can only be used once. We don't have to do full things like checking things like `X-Forwarded-By` or anything like that, but checking the host would make it so paths don't clobber each other (which feels likely, especially when people are testing things or kicking the tires)

Reply:
I saw path based routing as a very simple way to setup different paths for the scenario of a couple components handling HTTP requests, but it falls apart _pretty_ quickly as often you do want to handle multiple paths.

I was thinking it would be great to also have a `host` / `domain` based mode that allows you to route based on the host header, and not the path, since that would let you easily route all requests to a particular domain to a component. Thoughts? I think combining them might just make it overly complicated, since in the host-based mode I probably would just want all paths to go to the same component